python_modules/dagster/dagster/_core/definitions/asset_data_sla.py

```diff
@@ -0,0 +1,113 @@
+from typing import Mapping, Optional
+
+import pendulum
+from croniter import croniter
+
+from .events import AssetKey
+
+
+class RootDataTimeConstraint:
+    """Specifies a constraint on the timestamp of the root data that was used for the most recent
+    asset materialization of a given asset.
+
+    The root data timestamp is calculated by traversing the history of asset materializations of
+    upstream assets which occured before the current materialization. This gives a lower bound on
+    the most recent records that could possibly be incorporated into the current state of the asset
+    to which this SLA is attached.
+    """
+
+    def is_passing(self, root_data_timestamp: Optional[float]) -> bool:
```

Comment:
For purposes of testing and determining whether SLAs were met in the past, it might be helpful for this to accept a `current_time` argument, like `PartitionsDefinition.get_partitions` does.